### PR TITLE
Documentation overhaul

### PR DESCRIPTION
**Pull request by vyacheslav-pushkin**

### CODE REVIEW

File: packages/cuba-react-core/src/app/MainStore.ts [hunk 1]
```diff
@@ -7,17 +7,41 @@ export class MainStore {
...
+   * Currently selected locale.
+   */
   @observable locale?: string;
 
+  /**
+   * Information about project entities.
+   */
   @observable metadata?: IObservableArray<MetaClassInfo>;
+  // TODO can be made private?
   metadataRequestCount = 0;
```
Comment (vadimbasko) [inline packages/cuba-react-core/src/app/MainStore.ts:33]: I think we can resolve all TODO's in this file to make properties and methods private. Didn't see any cause not to do this. @web-devel ?

File: packages/cuba-react-core/src/app/MainStore.ts [hunk 2]
```diff
@@ -108,7 +152,9 @@ export class MainStore {
     return !this.authenticated && !this.usingAnonymously;
   }
 
-  /**
+  // TODO same method as `isEntityDataLoaded`, either one shall be removed.
+  // TODO Perhaps `isEntityDataLoaded` is a better name as this method can not only be used  in the entity management screens.
+  /*
    * Define id mainStore is ready to serve entiry management components.
```
Comment (vadimbasko) [inline packages/cuba-react-core/src/app/MainStore.ts:158]: Agree. This should be removed on merge paging. But I miss it. Could be replaced with `isEntityDataLoaded`.

File: packages/cuba-react-ui/src/ui/table/DataTable.tsx [hunk 3]
```diff
@@ -29,6 +29,9 @@ import {
 
 // todo may be it make sense to add 'defaultDataTableProps' const instead of describe each property in jsdoc,
 //  it's easy to forget update js docs after DataTable.defaultProps changes
```
Comment (vadimbasko) [inline packages/cuba-react-ui/src/ui/table/DataTable.tsx:31]: I think this TODO could be removed. Now we have comprehensive description for each property.

File: packages/cuba-react-core/src/app/MainStore.ts (hunk 1 above)
Comment (web-devel) [reply packages/cuba-react-core/src/app/MainStore.ts:33]: Agree

**Pull request merged**